src/lib.rs

```diff
@@ -33,33 +33,50 @@ use std::os::unix::prelude::*;
 use std::path::Path;
 
 use bytes::{Buf, BufMut};
-use futures::{Future, Poll, Async, Stream};
-use futures::sync::oneshot;
+use futures::{Async, Future, Poll, Stream};
+use tokio_io::{AsyncRead, AsyncWrite};
 use iovec::IoVec;
-use tokio_core::reactor::{PollEvented, Handle};
-#[allow(deprecated)]
-use tokio_core::io::Io;
-use tokio_io::{IoStream, AsyncRead, AsyncWrite};
+use tokio_reactor::{Handle, PollEvented};
+use mio::Ready;
 
 mod frame;
-pub use frame::{UnixDatagramFramed, UnixDatagramCodec};
+pub use frame::{UnixDatagramCodec, UnixDatagramFramed};
 mod ucred;
 pub use ucred::UCred;
 
-fn would_block() -> io::Error {
-    io::Error::new(io::ErrorKind::WouldBlock, "would block")
+/// Stream of listeners
+pub struct Incoming {
+    inner: UnixListener,
+}
+
+impl Stream for Incoming {
+    type Item = (UnixStream, SocketAddr);
+    type Error = io::Error;
+
+    fn poll(&mut self) -> Poll<Option<Self::Item>, io::Error> {
+        Ok(Some(try_nb!(self.inner.accept())).into())
+    }
 }
 
 /// A Unix socket which can accept connections from other unix sockets.
 pub struct UnixListener {
     io: PollEvented<mio_uds::UnixListener>,
-    pending_accept: Option<oneshot::Receiver<io::Result<(UnixStream, SocketAddr)>>>,
+    handle: Handle,
```

Comment:
It does, but I also need to create a new `PollEvented` when creating the stream where it seems to (at least from the code that were there before) that the same handle that were used to create the listener should be used to create the stream https://github.com/tokio-rs/tokio-uds/pull/36/files/f2ad0e079570f713dbcd82988a03c415ad67604c#diff-b4aea3e418ccdb71239b96952d9cddb6R152 .

Since `PollEvented` does not give a away to retrieve its handle I figured I need to store it here as well? (Otherwise only the current handle can be used?)

Reply:
It's by design that the accepted stream does not use the listener's `Handle`. This allows streams to be spread out across reactors instead of being forced to the same one.

You can also have a `poll_accept_std` that does no binding. See [here](https://github.com/tokio-rs/tokio/blob/master/tokio-tcp/src/listener.rs#L113).